sycl/source/interop_handle.cpp

```diff
@@ -0,0 +1,48 @@
+//==------------ interop_handle.cpp --- SYCL interop handle ----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl/detail/accessor_impl.hpp>
+#include <CL/sycl/interop_handle.hpp>
+#include <detail/context_impl.hpp>
+#include <detail/device_impl.hpp>
+#include <detail/queue_impl.hpp>
+
+#include <algorithm>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+
+pi_native_handle interop_handle::getNativeMem(detail::Requirement *Req) const {
+  auto Iter = std::find_if(std::begin(MMemObjs), std::end(MMemObjs),
+                           [=](ReqToMem Elem) { return (Elem.first == Req); });
+
+  if (Iter == std::end(MMemObjs)) {
+    throw("Invalid memory object used inside interop");
```

Comment:
This should probably be a `sycl::runtime_error` or `sycl::invalid_object_error`

Reply:
Done.